### ISSUE TITLE
TrueNAS API SMB won't provision volume when using API key

### ISSUE THREAD

**Issue by 93imagine** [democratic-csi/democratic-csi#416]
When creating a PVC the volume won't provision. 

k describe pvc name
`Changes to auxiliary parameters for SMB shares are restricted to users with full administrative privileges`

When using admin credentials for authentication instead of an API key. The volume provisions just fine.

Should be easy to reproduce.

**Comment by eaglesemanation** [comment]
This issue is still relevant, here is a nuclear option workaround. Do this carefully and be sure that you can physically access your NAS in case of failure, or at least over SSH, because this might bring WEB UI down:

- Check if `/usr` path is mounted as read-write. If it's read-only, do something similar to this:
```
zfs set readonly=off boot-pool/ROOT/x.x.x.x/usr
```
- Open `/usr/lib/python3/dist-packages/middlewared/plugins/smb.py` in your CLI editor of choice.
- Search for this piece of code:
```python
if app and not credential_has_full_admin(app.authenticated_credentials):
  if data['auxsmbconf']:
      verrors.add(
          'smb_update.auxsmbconf',
          'Changes to auxiliary parameters for SMB shares are restricted '
          'to users with full administrative privileges.'
      )
```
- Comment out this whole block.
- Set readonly back to `on`, just to avoid any accidents

I've looked into middlewared APIs, and while api_keys have some form of RBAC I couldn't figure out if I could explicitly whitelist `auxsmbconf` in there. The fix on democratic-csi side would probably be to stop using `auxsmbconf` parameter all together. As far as I understand, it's just being used to allow guest access to shares, didn't look into why that's needed.

**Comment by travisghansen** [comment]
Yeah, not much that can be done if the api disallows it. Feel free to open a ticket with iX and link it here so we can track it.
